src/ports/postgres/modules/deep_learning/madlib_keras.py_in

```diff
@@ -342,23 +330,24 @@ def compute_loss_and_metrics(schema_madlib, table, dependent_varname,
     given dataset (table.)
     """
     start_val = time.time()
-    evaluate_result = get_loss_acc_from_keras_eval(schema_madlib,
-                                                   table,
-                                                   dependent_varname,
-                                                   independent_varname,
-                                                   compile_params,
-                                                   model_arch, model_state,
-                                                   gpus_per_host,
-                                                   segments_per_host,
-                                                   seg_ids_val,
-                                                   rows_per_seg_val,
-                                                   gp_segment_id_col)
+    evaluate_result = get_loss_metric_from_keras_eval(schema_madlib,
+                                                      table,
+                                                      dependent_varname,
+                                                      independent_varname,
+                                                      compile_params,
+                                                      model_arch,
+                                                      model_state,
+                                                      gpus_per_host,
+                                                      segments_per_host,
+                                                      seg_ids_val,
+                                                      rows_per_seg_val,
+                                                      gp_segment_id_col)
     end_val = time.time()
     plpy.info("Time for evaluation in iteration {0}: {1} sec.". format(
         curr_iter, end_val - start_val))
-    if len(evaluate_result) < 2:
-        plpy.error('Calling evaluate on table {0} returned < 2 '
-                   'metrics. Expected both loss and a metric.'.format(
+    if len(evaluate_result) not in [1, 2]:
```

Comment:
Since metric can be an optional param, isn't it possible that keras.eval only returns the loss ?

Reply:
Addressed

Reply:
evaluate_result is always a list, and evaluate_result[0] is always loss, and we only allow one more metric here, so that the valid length of evaluate_result should either 1 or 2